application/lib/tutorial/six/module/chatView.dart

```diff
@@ -0,0 +1,147 @@
+import 'package:flutter/material.dart';
+import 'package:flutter/semantics.dart';
+import 'package:flutter/services.dart';
+import 'chatModel.dart';
+
+class ChatView extends StatefulWidget {
+  @override
+  _ChatViewState createState() => _ChatViewState();
+}
+
+class _ChatViewState extends State<ChatView>{
+  bool isFirstTime = true;
+  final TextEditingController textEditor = TextEditingController();
+  List<ChatModel> messages = [];
+
+  @override
+  void dispose(){
+    textEditor.dispose();
+    super.dispose();
+  }
+  Widget _buildTextField(){
+    return Container(
+      margin: const EdgeInsets.symmetric(horizontal: 8.0),
+      child: Row(
+        children: [
+          Flexible(
+            child: TextField(
+              controller: textEditor,
+              //onSubmitted: _handleSubmit,
+              decoration: InputDecoration.collapsed(
+                hintText: "Send a message",
+              ),
+            ),
+          ),
+          IconButton(
+            icon: Icon(Icons.send),
+            onPressed: () => _handleSubmit(textEditor.text, messages.length+1),
+          ),
+        ],
+      ),
+    );
+  }
+  void _handleSubmit(String text, int id) {
+    textEditor.clear();
+    bool containMessage = text.contains("bob");
+    ChatModel message = ChatModel(
+      message: text,
+      isMe: true,
+      id: id,
+    );
+    speakDuringLesson(message.message);
+    setState(() {
+      messages.add(message);
+      debugPrint("added message");
+    });
+    if (containMessage){
+      speakDuringLesson("you have successfully complete the lesson, now navigate back to tutorial 6");
+      Future.delayed(Duration(seconds: 8), () {
```

Comment:
This works for now, but only for user with default talkback speed. I'm still not sure what the best solution is for these. I believe the [flutter_tts](https://pub.dev/documentation/flutter_tts/latest/flutter_tts/FlutterTts/speak.html) plugin gives you the ability to await speech completion, but not sure how this plays with TalkBack.